tools/prepare_lava_job.py

```diff
@@ -0,0 +1,169 @@
+    nargs="?",
+    help="name of device type to submit job. Default: RevPi_Core",
+)
+
+parser.add_argument(
+    "-u",
+    "--url",
+    dest="url",
+    help="URL to the image",
+)
+
+parser.add_argument(
+    "--flash",
+    dest="flash",
+    nargs="?",
+    required=True,
+    help="file used to flash image",
+)
+
+parser.add_argument(
+    "-s",
+    "--testsuite",
+    dest="testsuite",
+    required=True,
+    help="job testsuite to be used after flashing",
+)
+
+parser.add_argument(
+    "-l",
+    "--lava",
+    dest="lava",
+    type=str,
+    nargs='?',
+    const="True",
+    help="submit generated job with lavacli",
+)
+
+def run_lavacli() -> bool:
+    try:
+        pkg_resources.get_distribution("lavacli")
+    except pkg_resources.DistributionNotFound:
+        print("lavacli is not installed...")
+        return False
+    
+    lavacli_cmd = LAVACLI_JOB_SUBMIT + NEW_JOB_NAME
+    process = subprocess.Popen(lavacli_cmd, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    stdout, stderr = process.communicate()
+    stdout = stdout.decode('utf-8')
+    stderr = stderr.decode('utf-8')
+    
+    print(f"{lavacli_cmd}")
+    print(stdout)
+
+    if stderr:
+        print("lavacli Error:")
+        print(stderr)
+        return False
+
+    return True
+
+#TODO what about another URLs???
+def get_imagename_from_url(url):
+    return os.path.basename(url)
+
+def add_namespace_in_testsuite(content):
+    actions = content.get("actions", [])
+
+    for action in actions:
+        if "test" in action or "boot" in action or "deploy" in action:
+            action_name = next(iter(action))
+            action[action_name]["namespace"] = "testsuite"
+
+    return content
+
+def create_job(device_type, image_url, file_flash, file_testsuite) -> bool:
+    print("Generate Job... ", end="")
+
+    # Load the contents of file_flash
+    try:
+        with open(file_flash, 'r') as file:
+            file_flash_contents = yaml.safe_load(file)
+    except OSError:
+        print(f"\nCould not read file to flash device: {file_flash}")
+        return False
+    print(file_testsuite)
+    # Load the contents of file_testsuite
+    try:
+        with open(file_testsuite, 'r') as file:
+            file_testsuite_contents = yaml.safe_load(file)
+    except OSError:
+        print(f"\nCould not read file with testsuite: {file_testsuite}")
+        return False
+    print("blablabla... ")
+    if device_type:
+        # Update device_type
+        file_flash_contents['device_type'] = device_type
+    
+    if not image_url:
+        image_url = file_flash_contents['actions'][2]['deploy']['images']['recovery_image']['url']
+
+    # Update job_name
+    file_flash_contents['job_name'] = 'Flash & Test Image: '+ get_imagename_from_url(image_url)
+    
+    if image_url:
+        # Update the image URL with the provided image parameter
+        file_flash_contents['actions'][2]['deploy']['images']['recovery_image']['url'] = image_url
```

Comment:
Indexing the `actions` array like that is fine for now, but we might need to make this more dynamic at some point so that the `deploy` action doesn't have to be at index `2`.